src/hl/container.rs

```diff
@@ -346,6 +347,115 @@ impl<'a> Writer<'a> {
     }
 }
 
+#[derive(Debug)]
+pub struct ByteReader<'a> {
+    obj: &'a Container,
+    pos: u64,
+    dt: Datatype,
+    obj_space: Dataspace,
+    xfer: PropertyList,
+}
+
+impl<'a> ByteReader<'a> {
+    pub fn new(obj: &'a Container) -> Result<Self> {
+        ensure!(!obj.is_attr(), "ByteReader cannot be used on attribute datasets");
+
+        let file_dtype = obj.dtype()?;
+        let mem_dtype = Datatype::from_type::<u8>()?;
+        file_dtype.ensure_convertible(&mem_dtype, Conversion::NoOp)?;
+
+        let obj_space = obj.space()?;
+        ensure!(obj_space.shape().len() == 1, "Only rank 1 datasets can be read vie ByteReader.");
+        let xfer = PropertyList::from_id(h5call!(H5Pcreate(*crate::globals::H5P_DATASET_XFER))?)?;
+        if !hdf5_types::USING_H5_ALLOCATOR {
+            crate::hl::plist::set_vlen_manager_libc(xfer.id())?;
+        }
+        Ok(ByteReader { obj, pos: 0, obj_space, dt: mem_dtype, xfer })
+    }
+
+    pub fn set_position(&mut self, pos: u64) {
```

Comment:
Hm I just added this as an equivalent to io::Cursor where it is public. But I guess, using the seek interface is better. I could also remove this method.

Reply:
If you want to make it like `io::Cursor`, then I guess there should also be a public `position()` method. I'm not sure whether copying its interface is useful, but I guess it may be

Reply:
I removed it for now. If we need it, we can add it later.